### DISCUSSION TITLE
Object initializer syntax: anonymous types vs actual types

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#2665] [Language Ideas]
Right now I an create an anonymous type like so:
````
            var Start = 1;
            var B = new {
                Start,
                End = 100
            };
````

When I create a regular type, I can't do that though and have to specify the property name even when it matches a variable name.  I should be able to do:
````
            var Start = 1;
            var B = new Range() {
                Start, //Currently I have to do: Start = Start
                End = 100
            };
````

**Comment by iam3yal** [comment]
@TonyValenti I feel like you shouldn't because in the first example you define a new property which makes more sense to me whereas in the second example you initialize an existing property so I don't know, it might be confusing to some people but then maybe it's just me.

Edit: Rephrased.

**Comment by quinmars** [comment]
That reminds me a little bit on #2616

```csharp
        int Capacity = 100;
        var list = new List<int> ()
        {
            Capacity
        };
```

This compiles and adds the value 100 to the list.
